cabal-install-solver/src/Distribution/Solver/Types/PkgConfigDb.hs

```diff
@@ -65,9 +67,14 @@ readPkgConfigDb verbosity progdb = handle ioErrorHandler $ do
         -- The output of @pkg-config --list-all@ also includes a description
         -- for each package, which we do not need.
```

Comment:
As witnessed by
- https://github.com/actions/runner-images/issues/6364

`pkg-config` might be present but `pkg-config --list-all` might segfault.  What is the strategy in such situations?  (Maybe building a pkg-config database eagerly isn't so great after all, one could fill it lazily by demand in the solver, could one?)

Reply:
It looks like cabal treats an error in `pkg-config --list-all` similarly to the inability to find pkg-config, so the solver would only fail in the case where all potential solutions require at least one pkg-config package.

The solver is currently pure, so querying packages on demand would be a big change.

Reply:
Programming around "an external program is segfaulting" seems like its really pushing it, though I do understand a frustration with images being broken. I would again advise that text-icu put in a flag that varies depending on pkg-config being present and working or not.

Reply:
Yeah, but I'd like to know whether `pkg-config --list-all` works at all on `macOS-11/12`.  Unfortunately, I cannot test it directly as I my machine is still on `macOS-10`.  Anyone with a recent macOS willing to investigate thie?

Reply:
So, upstream came back with an analysis: https://github.com/Homebrew/homebrew-core/issues/112785
Not that the details matter, but the upshot is that _one_ broken package configuration breaks `pkg-config --list-all` and consequently `cabal-3.8.1.0`. 
So, install `imagemagick` or `highway` and suddently your haskell packages fail to build with cabal.  Nightmare scenario?  Yes!
Lesson to learn: stay away from the `pkg-config --list-all` path and only query `pkg-config` for the packages you are interested in.

Reply:
Another idea is to work around the segfault by not enforcing pkg-config dependencies when pkg-config exists but the --list-all command fails. Not enforcing pkg-config dependencies is the same behavior as before #7621.

I also think that we should merge this PR first, since it already handles one case where a pkg-config command fails.

Reply:
I think a broken pkg-config is the same situation as a missing one -- a hard fail is fine, because the intended user pattern should be that there is an auto flag in the package that can switch to how to handle the case where pkg-config is unavailable (because non-existent or broken).

I am aware that not all packages have that flag that allows this switching. But we should encourage them to add it!